### ISSUE TITLE
Upgrade Cabal 3.4.0.0

### ISSUE THREAD

**Issue by brandon-leapyear** [commercialhaskell/stack#5558]
Blocked by https://github.com/commercialhaskell/pantry/pull/38

Support Cabal 3.4.0.0; _might_ resolve https://github.com/commercialhaskell/stack/issues/4428 and https://github.com/commercialhaskell/stack/issues/5284

**Comment by kk-hainq** [comment]
I would like to add that Cabal 3.4 seems to fix https://github.com/haskell/cabal/issues/6470, which makes #5318 a lot easier to handle from `stack`'s side.

**Comment by brandon-leapyear** [comment]
@kk-hainq see the linked PR. Blocked on someone on stack's side reviewing, helping with CI failures, etc

**Comment by hasufell** [comment]
I think this was recently fixed by @mpilgrem

**Comment by mpilgrem** [comment]
From commit a9f00ffe4908b7f55af938f5a85943859f1e7af4 (1 Apr 2022), the `stack` executable can be built with `lts-19.x` (GHC 9.0.2, `Cabal-3.4.1.0`). I have a separate pull request on moving Stack to GHC 9.0.2 (#5736) - a related but different task - which is currently stuck on the CI for Unix-like operating systems. That may be overtaken by the intention of @psibi to move Stack to GHC 9.2.3 (#5763 and https://github.com/commercialhaskell/stack/pull/5736#issuecomment-1152541872).

**Comment by brandon-leapyear** [comment]
Seems like this PR is redundant as of #5763
